src/test/java/net/snowflake/client/jdbc/SnowflakeUtilTest.java

```diff
@@ -93,7 +93,8 @@ private static SnowflakeColumnMetadata createExpectedMetadata(
             rootNode.path("database").asText(),
             rootNode.path("schema").asText(),
             rootNode.path("table").asText(),
-            false);
+            false,
+            rootNode.path("dimension").asInt());
     return expectedColumnMetadata;
   }
 
```

Comment:
Correct me if I'm wrong, but I believe that this change does not change much from the customer's perspective :) I believe, that what a better solution is is to propagate `dimension` to `SnowflakeResultSetMetaDataV1` and expose it via `unwrap`. Then it will be accessible in metadata of a resultSet instead of querying connection metadata. WDYT?

Reply:
It's possible but needs the unwrapping whereas we can use the current `getColumns` with passing `COLUMN_SIZE` value (check the added test) - it's much easier to use for the caller